### PR TITLE
Fix/361 radio button

### PR DESCRIPTION
**Pull request by harshil1793**
## Description

This PR fixes the following key items for RadioGroup component:

- If radio button options are boolean then formik state will store it as boolean
- Navigating after selecting option in radio button, value will persist in UI
- Changes to the pages that are using formik state values as strings

## Original issue(s)
department-of-veterans-affairs/va-forms-system-core#361

## Screenshots
img 1
img 2

## Definition of done
- [ ] Documentation has been updated, if applicable
- [ ] A link to the original github issue has been provided
- [ ] No sensitive information (i.e. PII/credentials/internal URLs etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: examples/burial-form/PlotAllowance.jsx [hunk 1]
```diff
@@ -28,7 +28,7 @@ export default function PlotAllowance(props) {
               ]
             }
           />
-          { formikContext?.values?.federalCemetery === "false" && (
```
Comment (mlauren) [inline examples/burial-form/PlotAllowance.jsx:31]: this should be formikContext?.values?.federalCemetery === false

Comment (harshil1793) [reply examples/burial-form/PlotAllowance.jsx:31]: I have added '!' which will is identical to what you're suggesting.

File: examples/burial-form/PlotAllowance.jsx [hunk 2]
```diff
@@ -16,7 +16,7 @@ export default function PlotAllowance(props) {
     <div className="vads-u-margin-x--1p5">
       <Page {...props} nextPage="/claimant-contact-information" prevPage="/benefits/burial-allowance">
         <TextField required name="placeOfRemains" label="Place of burial or deceased Veteran’s remains" />
-        <div className={'vads-u-padding-y--1p5 form-expanding-group' + (formikContext?.values?.federalCemetery === "false" && ' form-expanding-group-open')}>
```
[the remark this answers is not in the document]
Comment (harshil1793) [reply examples/burial-form/PlotAllowance.jsx:19]: I have added '!' which will is identical to what you're suggesting.

File: src/form-builder/RadioGroup.tsx [hunk 3]
```diff
@@ -21,7 +21,25 @@ export function RadioGroup(props: RadioGroupProps): JSX.Element {
...
+      case 'false':
+      case 'no':
+      case '0':
+      case null:
+        return false;
+      default:
+        return Boolean(value);
+    }
+  };
+
```
[the remark this answers is not in the document]
Comment (harshil1793) [reply src/form-builder/RadioGroup.tsx:42]: yea that's a good point

**Review by mlauren — APPROVED** [review]

**Comment by mlauren** [comment]
LGTM :) thanks

**Pull request merged**